### PR TITLE
refactor(analytics): send JSON in filter change events

### PR DESCRIPTION
**Pull request by anandaroop**
https://artsyproduct.atlassian.net/browse/FX-2850
↳ https://artsyproduct.atlassian.net/browse/FX-2913

Follow-up to https://github.com/artsy/cohesion/pull/184

Filter events on all artwork grids should now send their `current` and `changed` properties as JSON blobs, rather than as nested objects, which cause problems over in Redshift.

So this bumps Cohesion to v2 and —

- Uses the updated API for Cohesion's `commercialFilterParamsChanged` helper
- Update non-Cohesion events from auction results & auction artworks grid

This is how they look in Segment —

### Standard artwork grids, via Cohesion (artist, artist series, fair, show, etc)

regular grid

### Auction artworks grid

auction grid

### Auction results

auction results

### CODE REVIEW

**Comment by anandaroop** [comment]
cc @louislecluse @mikehrom

File: src/desktop/apps/auction/utils/__tests__/analyticsMiddleware.jest.ts [hunk 1]
```diff
@@ -33,13 +33,13 @@ describe("analytics middleware", () => {
         auction_slug: "auction-slug",
         sale_id: "123",
         user_id: "321",
-        current: [
+        current: JSON.stringify([
```
Comment (damassi) [inline src/desktop/apps/auction/utils/__tests__/analyticsMiddleware.jest.ts:36]: Makes sense

**Review by damassi — APPROVED** [review]
🟢

**Comment by anandaroop** [comment]
@louislecluse @mikehrom this is on staging now, if you want to confirm that the correct events are being fired from staging.artsy.net

**Pull request merged**